src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/uploadwelshdocument/UploadWelshDocumentsSubmittedHandler.java

```diff
@@ -38,39 +41,73 @@ public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callb
 
         return callbackType.equals(CallbackType.SUBMITTED)
                 && callback.getEvent().equals(EventType.UPLOAD_WELSH_DOCUMENT)
-                && !callback.getCaseDetails().getState().equals(State.INTERLOCUTORY_REVIEW_STATE)
-                && StringUtils.isNotEmpty(callback.getCaseDetails().getCaseData().getSscsWelshPreviewNextEvent());
+                && !callback.getCaseDetails().getState().equals(State.INTERLOCUTORY_REVIEW_STATE);
```

Comment:
Any particular reason for removing the getSscsWelshPreviewNextEvent()? I think its being called next event after the welsh document is uploaded

Reply:
@srochani - I've changed this to include the check. Originally the reinstatement request was not going to have a nextEvent associated with it hence removing it. But now it uses Update_Case_only so added it back in